src/arrayMethodSort.js

```
@@ -1,12 +1,72 @@
 'use strict';
 
-/**
- * Implement method Sort
- */
-function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+[].__proto__.sort2 = function (compareFunction) {
+  const arr = this;
+  const len = arr.length;
+
+  const defaultComparator = (a, b) => {
+    const aIsUndefined = a === undefined;
+    const bIsUndefined = b === undefined;
+
+    if (aIsUndefined && bIsUndefined) {
+      return 0;
+    }
+
+    if (aIsUndefined) {
+      return 1;
+    }
+
+    if (bIsUndefined) {
+      return -1;
+    }
+
+    const aStr = String(a);
+    const bStr = String(b);
+
+    if (aStr < bStr) {
+      return -1;
+    }
+
+    if (aStr > bStr) {
+      return 1;
+    }
+
+    return 0;
   };
-}
 
-module.exports = applyCustomSort;
+  const comparator =
+    typeof compareFunction === 'function' ? compareFunction : defaultComparator;
```

This line sets `comparator` using `typeof compareFunction === 'function' ? compareFunction : defaultComparator`. That means if the caller passes a non-function value (for example `null` or a number) your implementation will silently fall back to the default comparator. According to `Array.prototype.sort` semantics (MDN/spec), if `compareFn` is provided and is not a function, a `TypeError` should be thrown. Consider explicitly checking and throwing, for example:

```javascript
if (compareFunction !== undefined && typeof compareFunction !== 'function') {
  throw new TypeError('The comparison function must be a function');
}
const comparator = typeof compareFunction === 'function' ? compareFunction : defaultComparator;
```

This will make `sort2` consistent with the native `Array.prototype.sort` behavior.